api/src/pcapi/core/external_bookings/decorators.py

```diff
@@ -0,0 +1,91 @@
+import functools
+import inspect
+import logging
+from typing import Any
+from typing import Callable
+from typing import TypeVar
+
+from pcapi.utils.requests import exceptions as requests_exception
+
+from . import exceptions
+from . import models
+
+
+logger = logging.getLogger(__name__)
+
+
+class ExternalBookingDecoratorException(Exception):
+    pass
+
+
+def _extract_logging_information(client_func: Callable, *args: list[Any], **kwargs: Any) -> dict:
+    """
+    Return a dictionary to be passed to the `extra` parameter of the logging function.
+
+    The returned dictionary contains the following keys:
+        - "cinema_id" (int): The ID of the cinema from the pivot table.
+        - "client" (str): The client used for the request (e.g., `EMSClientAPI`).
+        - "method" (str): The instance method used for the request (e.g., `book_ticket`).
+        - "method_params" (dict): The parameters passed to the method. Example:
+            {
+                "show_id": "12345",             # ID of the show
+                "booking": "<Booking #1>",      # Stringified Booking instance
+                "beneficiary": "<User #3>",     # Stringified User instance
+            }
+    """
+    self = args[0]
+    assert isinstance(self, models.ExternalBookingsClientAPI)  # to make mypy happy
+    # as client_func is an instance method of `ExternalBookingsClientAPI`, self should have a `cinema_id`
+    cinema_id = self.cinema_id
+
+    func_parameters = [param.name for param in inspect.signature(client_func).parameters.values()]
+    method_params = {}
+
+    arg_index = 1  # as the first argument is `self`, start from index 1
+    for arg_value in args[1:]:
+        arg_name = func_parameters[arg_index]
+        method_params[arg_name] = str(arg_value)
+        arg_index += 1
+
+    for k, v in kwargs.items():
+        method_params[k] = str(v)
```

Comment:
```suggestion
    method_params = inspect.signature(client_func).bind(*args, **kwargs).arguments
    method_params = {name: str(arg) for name, arg in method_params.items()}
    method_params.pop('self')
```

Ce qui renvoie par exemple : 

```python
{'show_id': '12345', 'booking': '<Booking #1>', 'beneficiary': '<User #3>'}
```

Et les tests passent chez moi. Ce changement implique d'ignorer ma suggestion précédente vis-à-vis de `self` et de `args`.

Qu'en dis-tu ?

Reply:
Ah cool parfait

Reply:
Je vais utiliser ça